tools/cldr-code/src/test/java/org/unicode/cldr/unittest/TestUtilities.java

```diff
@@ -1409,4 +1372,20 @@ public void TestBulkUploadHtml() {
         assertEquals("writeBulkInfoHtml", expected, out.toString());
     }
 
+    /**
+     * Verify that VettingViewer.getMissingStatus returns MissingStatus.PRESENT
+     * for a typical path in a well-populated locale
+     *
+     * Ideally we should also test for MissingStatus.DISPUTED, etc.; that's more difficult
+     */
```

Comment:
How could we test for MissingStatus.DISPUTED? It would require the kind of CLDRFile that reflects voting in progress; do we have a way to set that up in the context of cldr-unittest TestAll?

Reply:
We'd have to set it up so that the callback returned the interesting information, I'm guessing.

Reply:
One way would be to leverage the `TestSTFactory` which already sets up vetters, disputes, etc. It actually creates a STFactory though which has a DB backing.

But all you need is a `UsersChoice<Organization>` subclass - `VettingViewer.main()` constructs one which includes disputed values, which looks quite suitable for a test case.

Reply:
actually, `getMissingStatus()` will never return `disputed`, but member function `getErrorOnPath()` will.

Reply:
> actually, `getMissingStatus()` will never return `disputed`, but member function `getErrorOnPath()` will.

Good point! The ancient bug, for which this is a follow-up, was https://unicode-org.atlassian.net/browse/CLDR-11765 and it involved items under Disputed in the Dashboard, hence my confusion...

I think this is the first unit test for any method in VettingViewer. Obviously more tests would be helpful.

I doubt whether getMissingStatus even belongs in VettingViewer -- shouldn't the status be the same regardless of whether an item is viewed in the Dashboard or the DataSection view?

Reply:
@btangmu it seems to be more of a utility function that may belong somewhere else.